quickwit-index-config/src/query_builder.rs

```diff
@@ -49,6 +49,43 @@ pub(crate) fn build_query(
     Ok(query)
 }
 
+/// Extracts all filters related to tag fields.
+/// Returns a list of `tag_field_name:tag_value` found in the query.
+pub fn extract_tags_from_query(
+    raw_query: &str,
+    tag_field_names: &[String],
+) -> Result<Vec<String>, QueryParserError> {
+    let user_input_ast = tantivy_query_grammar::parse_query(raw_query)
+        .map_err(|_| TantivyQueryParserError::SyntaxError)?;
+
+    Ok(collect_tag_filters(user_input_ast, tag_field_names))
+}
+
+fn collect_tag_filters(user_input_ast: UserInputAst, tag_field_names: &[String]) -> Vec<String> {
+    let mut fields = vec![];
+    match user_input_ast {
+        UserInputAst::Clause(sub_queries) => {
+            for (_, sub_ast) in sub_queries {
```

Comment:
The `_` is actually important, the Occur information.
That's what decides whether the term should be or must be here. (NOT is also possible.

In the boolean logic world, is it A AND B or A OR B.